fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/PlanHashable.java

```diff
@@ -114,6 +114,9 @@ static int objectPlanHash(@Nonnull final PlanHashKind hashKind, @Nullable Object
         if (obj instanceof Iterable<?>) {
             return iterablePlanHash(hashKind, (Iterable<?>)obj);
         }
+        if (obj.getClass().isArray()) {
+            return iterablePlanHash(hashKind, Arrays.asList((Object[])obj));
+        }
```

Comment:
Done.
Unfortunately, had to introduce specific handling for primitive arrays, since obj.hashCode for an array (as well as Objects.hashCode) do not generate a stable value. Added a test class for the static utilities in PlanHashable.